test/libyul/yulOptimizerTests/fullSuite/devcon_example.yul

```diff
@@ -21,8 +21,9 @@
 //     {
 //         let _1 := calldataload(0)
 //         let sum := 0
+//         let length := calldataload(_1)
 //         let i := sum
-//         for { } lt(i, calldataload(_1)) { i := add(i, 1) }
+//         for { } lt(i, length) { i := add(i, 1) }
```

Comment:
Wht is `length` not rematerialized here?

Reply:
Because def-use crosses a loop boundary.

Reply:
Ah okay, I guess I got confused with `origin()` being rematerialized in spite of crossing loop boundary. Now I realize that was because of `cost == 0` rule that @ekpyron pointed out